crates/ethportal-api/src/types/execution/header_with_proof.rs

```diff
@@ -3,36 +3,53 @@ use jsonrpsee::core::Serialize;
 use serde::Deserialize;
 use ssz::SszDecoderBuilder;
 use ssz_derive::{Decode, Encode};
-use ssz_types::{typenum, FixedVector, VariableList};
+use ssz_types::{typenum, FixedVector};
 use tree_hash::TreeHash;
 
-use crate::types::{
-    bytes::ByteList1024,
-    consensus::{
-        beacon_block::{BeaconBlockBellatrix, BeaconBlockCapella},
-        beacon_state::{BeaconStateCapella, HistoricalBatch},
-        proof::build_merkle_proof_for_index,
-    },
-    execution::{
-        block_body::{MERGE_TIMESTAMP, SHANGHAI_TIMESTAMP},
-        ssz_header,
+use crate::{
+    consensus::{beacon_block::BeaconBlockDeneb, beacon_state::BeaconStateDeneb},
+    types::{
+        bytes::ByteList1024,
+        consensus::{
+            beacon_block::{BeaconBlockBellatrix, BeaconBlockCapella},
+            beacon_state::{BeaconStateCapella, HistoricalBatch},
+        },
+        execution::ssz_header,
     },
 };
 
+/// The timestamp of the first Merge block (block number: 15537394)
+pub const MERGE_TIMESTAMP: u64 = 1663224179;
+
+/// The timestamp of the first Shapella (Shanghai-Capella) slot.
+///
+/// - Slot: 6209536
+/// - Epoch: 194048
+/// - Block number: 17034870
+///     - Note that frst Shapella block is created at slot 6209538 (timestamp: 1681338479)
+pub const SHAPELLA_TIMESTAMP: u64 = 1681338455;
+
+/// The timestamp of the first Dencun (Cancun-Deneb) slot.
+///
+/// - Slot: 8626176
+/// - Epoch: 269568
+/// - Block number: 19426587
+pub const DENCUN_TIMESTAMP: u64 = 1710338135;
```

Comment:
One thing to note is long term instead of using these constants we would be required to pass NetworkSpec which would hold the fork timestamps per network mainnet,sepolia etc

Reply:
So in short going forward if we want to support testnets we won't be able to use any global consts for forks

Reply:
I will make a followup PR for using `NetworkSpec` instead of const's. I don't think I have concerns with this PR anymore

Reply:
Here is a PR for it https://github.com/ethereum/trin/pull/1780